src/r1cs/sparse.rs

```diff
@@ -85,6 +87,30 @@ impl<F: PrimeField> SparseMatrix<F> {
     }
   }
 
+  /// Samples a new random matrix of size `rows` x `cols` with `num_entries` non-zero entries.
+  pub fn random<R: RngCore + CryptoRng>(
+    rows: usize,
+    cols: usize,
+    num_entries: usize,
+    mut rng: &mut R,
+  ) -> Self {
+    assert!(num_entries <= rows * cols);
+
+    let mut indices = BTreeSet::<(usize, usize)>::new();
+    while indices.len() < num_entries {
+      let row = rng.next_u32() as usize % rows;
+      let col = rng.next_u32() as usize % cols;
+      indices.insert((row, col));
+    }
+
+    let matrix = indices
+      .into_iter()
+      .map(|(row, col)| (row, col, F::random(&mut rng)))
+      .collect::<Vec<_>>();
```

Comment:
I think this may (occasionally) produce duplicates, which is worrisome, because I think the `SparseMatrix` logic does not check for duplicates. That later remark is something we should test, BTW.

Summoning @winston-h-zhang to check what I'm saying.

Reply:
Edit: my bad, the `BTreeSet` guarantees this will not produce duplicates, and the while loop guarantees that this will produce exactly `num_entries`,  essentially doing rejection sampling. I'm turning blind.